### PR TITLE
Add Interac refund canceled event on back press while Interac refund is in progress

### LINKED ISSUE #6372: Add `*_interac_refund_cancelled` event
Track Interac payment canceled event. Interac payment canceled can happen when the merchant presses back on the device when the refund is in progress. 
`card_present_collect_interac_payment_canceled`

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Closes: #6372 

### Description

This PR adds `*_interact_refund_cancelled` event. This event gets fired up on back press while Interac refund is in progress.

### Testing instructions

1. Collect payment from an order eligible for IPP (store must be setup in Canada)
2. Make sure the payment is successfully collected from the above order.
3. Try to refund using the same Interac card for the above order.
4. While the Interac refund is in progress, press back and check the logs
5. Ensure `*_interact_refund_cancelled` event is triggered with appropriate properties.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/cardreader/payment/CardReaderPaymentViewModel.kt [hunk 1]
```diff
@@ -566,16 +565,18 @@ class CardReaderPaymentViewModel
...
+                } else if (isStateEligibleForTrackingInteracRefund(state)) {
+                    tracker.trackInteracRefundCancelled(getCurrentInteracRefundState())
                 }
             }
             triggerEvent(Exit)
         }
     }
 
-    private fun isStateEligibleForTracking(paymentState: ViewState) =
+    private fun isStateEligibleForTrackingPayment(paymentState: ViewState) =
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/cardreader/payment/CardReaderPaymentViewModel.kt]: Would it make sense to add

```
interface PaymentFlow
interface RefundFlow
```

And mark all the Payment Flow states and Refund with those interfaces and then check `paymentState is PaymentFlow` and the same for a refund?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/cardreader/payment/CardReaderPaymentViewModel.kt [hunk 2]
```diff
@@ -594,6 +595,23 @@ class CardReaderPaymentViewModel
         }
     }
 
+    private fun isStateEligibleForTrackingInteracRefund(refundState: ViewState) =
+        refundState is RefundLoadingDataState ||
+            refundState is CollectRefundState ||
+            refundState is ProcessingRefundState
+
+    private fun getCurrentInteracRefundState(): String? {
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/cardreader/payment/CardReaderPaymentViewModel.kt]: Following the idea from [this comment](https://github.com/woocommerce/woocommerce-android/pull/6394/files#r859584584), we could make it more declarative with something like that:

```

    interface PaymentFlowTrackableCancellation {
        val nameForTracking: String
    }

    object LoadingDataState : ViewState(
        headerLabel = R.string.card_reader_payment_collect_payment_loading_header,
        hintLabel = R.string.card_reader_payment_collect_payment_loading_hint,
        paymentStateLabel = R.string.card_reader_payment_collect_payment_loading_payment_state,
        isProgressVisible = true
    ), PaymentFlowTrackableCancellation {
        override val nameForTracking: String
            get() = "Loading"
    }
```

**Review by kidinov — COMMENTED** [review]
Looks and tests good! I left optional comments about potential code simplifications, let me know your thoughts!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/cardreader/payment/CardReaderPaymentViewModel.kt (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/cardreader/payment/CardReaderPaymentViewModel.kt]: Makes sense! How does this look? https://github.com/woocommerce/woocommerce-android/pull/6394/commits/9ff145d72180760d2efddd1c1a8f392618b35420

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/cardreader/payment/CardReaderPaymentViewModel.kt (hunk 2 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/cardreader/payment/CardReaderPaymentViewModel.kt]: https://github.com/woocommerce/woocommerce-android/pull/6394/commits/9ff145d72180760d2efddd1c1a8f392618b35420

**Review by kidinov — APPROVED** [review]

**Pull request merged**